Wrappers/Python/cil/framework/labels.py

```diff
@@ -0,0 +1,174 @@
+    HORIZONTAL_Y = "horizontal_y"
+
+    @classmethod
+    def get_default_order_for_engine(cls, engine):
+        engine_orders = {
+            Backends.ASTRA.value: [cls.CHANNEL.value, cls.VERTICAL.value, cls.HORIZONTAL_Y.value, cls.HORIZONTAL_X.value],
+            Backends.TIGRE.value: [cls.CHANNEL.value, cls.VERTICAL.value, cls.HORIZONTAL_Y.value, cls.HORIZONTAL_X.value],
+            Backends.CIL.value: [cls.CHANNEL.value, cls.VERTICAL.value, cls.HORIZONTAL_Y.value, cls.HORIZONTAL_X.value]
+        }
+        Backends.validate(engine)
+        engine = Backends.get_enum_value(engine)
+
+        return engine_orders[engine]
+
+    @classmethod
+    def get_order_for_engine(cls, engine, geometry):
+
+        dim_order = cls.get_default_order_for_engine(engine)
+        dimensions = [label for label in dim_order if label in geometry.dimension_labels ]
+
+        return dimensions
+
+    @classmethod
+    def check_order_for_engine(cls, engine, geometry):
+        order_requested = cls.get_order_for_engine(engine, geometry)
+
+        if order_requested == list(geometry.dimension_labels):
+            return True
+        else:
+            raise ValueError(
+                "Expected dimension_label order {0}, got {1}.\nTry using `data.reorder('{2}')` to permute for {2}"
+                .format(order_requested, list(geometry.dimension_labels), engine))
+        
+class AcquisitionDimensionLabels(_LabelsBase):
+    CHANNEL = "channel"
+    ANGLE = "angle"
+    VERTICAL = "vertical"
+    HORIZONTAL = "horizontal"
+
+    @classmethod
+    def get_default_order_for_engine(cls, engine):
+        engine_orders = {
+            Backends.ASTRA.value: [cls.CHANNEL.value, cls.VERTICAL.value, cls.ANGLE.value, cls.HORIZONTAL.value],
+            Backends.TIGRE.value: [cls.CHANNEL.value, cls.ANGLE.value, cls.VERTICAL.value, cls.HORIZONTAL.value],
+            Backends.CIL.value: [cls.CHANNEL.value, cls.ANGLE.value, cls.VERTICAL.value, cls.HORIZONTAL.value]
+        }
+        Backends.validate(engine)
+        engine = Backends.get_enum_value(engine)
+
+        return engine_orders[engine]
+
+    @classmethod
+    def get_order_for_engine(cls, engine, geometry):
+
+        dim_order = cls.get_default_order_for_engine(engine)
+        dimensions = [label for label in dim_order if label in geometry.dimension_labels ]
+
+        return dimensions
+
+    @classmethod
+    def check_order_for_engine(cls, engine, geometry):
+        order_requested = cls.get_order_for_engine(engine, geometry)
+
+        if order_requested == list(geometry.dimension_labels):
+            return True
+        else:
+            raise ValueError(
+                "Expected dimension_label order {0}, got {1}.\nTry using `data.reorder('{2}')` to permute for {2}"
+                .format(order_requested, list(geometry.dimension_labels), engine))
+    
+class FillTypes(_LabelsBase):
+    RANDOM = "random"
+    RANDOM_INT = "random_int"
+
+class UnitsAngles(_LabelsBase):
```

Comment:
My preference would be AngleUnits

Reply:
UnitsAngles means that if we add Distance or Energy in the future we can preface them all with units which will keep them grouped. i.e. 'UnitsDistance' 'UnitsEnergy'

Reply:
I think this one especially will only be used internally.